src/LogRecord.cpp

```diff
@@ -63,8 +53,13 @@ LogRecord::~LogRecord()
 
 std::ostringstream& LogRecord::initializeStream(const string& file, const int& line, const Log::LOG_LEVEL& level)
 {
-    m_stream.imbue(std::locale(m_stream.getloc(), new time_facet(g_sTimestampFormat.c_str())));
-    m_stream << microsec_clock::local_time() << " ["<<stripDirectory(file)<<":"<<line<<", "<<Log::logLevelToString(level);
+    auto now = std::chrono::system_clock::now();
```

Comment:
Is there any performance consequence in changing from boost to std::chrono ?

Generally we (quasar team) were super careful in that regard so that logging didn't affect performance more than it needed to.

Would it be possible to do a quick profile check - so a command like app where we log like 10^6 lines with the old and the new version and measure the time difference ?

Reply:
Actually, given that LogIt's main job is as part of Quasar servers, we should build a quick server on windows and linux and log from that. I can help to set this up with you

Reply:
> Is there any performance consequence in changing from boost to std::chrono ?
> 
> Generally we (quasar team) were super careful in that regard so that logging didn't affect performance more than it needed to.
> 
> Would it be possible to do a quick profile check - so a command like app where we log like 10^6 lines with the old and the new version and measure the time difference ?

Hi Ben,

TLDR: STD is almost 3 times faster than the Boost implementation

I just assembled a small program to calculate the performance of the new code over a million iterations.

CMakelists.txt
```
cmake_minimum_required(VERSION 3.10)
project(main_project)

# Set the C++ standard
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED True)

# Specify the path to Boost
set(BOOST_ROOT "/opt/3rdPartySoftware/boost/boost_1_81_0/")
set(Boost_INCLUDE_DIR "${BOOST_ROOT}/")
set(Boost_LIB_DIR "${BOOST_ROOT}/stage/lib")

# Find the Boost libraries
find_package(Boost 1.81.0 REQUIRED)

# Include Boost headers
include_directories(${Boost_INCLUDE_DIRS})

# Add the main executable
add_executable(main main.cc)

find_package(Boost 1.81.0 REQUIRED COMPONENTS date_time )

# Link Boost libraries if needed (example: Boost system and filesystem)
target_link_libraries(main Boost::date_time )

```

main.cc
```
#include <boost/date_time/posix_time/posix_time.hpp>
#include <boost/date_time/posix_time/posix_time_io.hpp>
#include <boost/format.hpp>

using boost::posix_time::time_facet;
using boost::posix_time::microsec_clock;

#include <chrono>
#include <iomanip>
#include <iostream>	

void test_std() {
    const std::string g_sTimestampFormat = "%Y-%m-%d %H:%M:%S";
    const auto now = std::chrono::system_clock::now();
    const auto now_time_t = std::chrono::system_clock::to_time_t(now);
    const auto now_us = std::chrono::duration_cast<std::chrono::microseconds>(now.time_since_epoch()) % 1000000;
    std::cout << std::put_time(std::localtime(&now_time_t), g_sTimestampFormat.c_str()) 
        << '.' << std::setw(6) << std::setfill('0') << now_us.count()
        << " Hello World!" << std::endl;
}

void test_boost() {
    const std::string g_sTimestampFormat = "%Y-%m-%d %H:%M:%s";
    std::cout.imbue(std::locale(std::cout.getloc(), new time_facet(g_sTimestampFormat.c_str())));
    std::cout << microsec_clock::local_time() << " Hello World!" << std::endl;
}

int main() {
    const int iterations = 1000000;

    auto start_std = std::chrono::high_resolution_clock::now();
    for (int i = 0; i < iterations; ++i) {
        test_std();
    }
    auto end_std = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed_std = end_std - start_std;

    auto start_boost = std::chrono::high_resolution_clock::now();
    for (int i = 0; i < iterations; ++i) {
        test_boost();
    }
    auto end_boost = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed_boost = end_boost - start_boost;

    std::cout << "Time elapsed for test_std: " << elapsed_std.count() << " seconds\n";
    std::cout << "Time elapsed for test_boost: " << elapsed_boost.count() << " seconds\n";

    return 0;
}
```

I repeated the tests 3 times and here are the results:

```
Time elapsed for test_std: 5.3959 seconds
Time elapsed for test_boost: 14.3323 seconds

Time elapsed for test_std: 5.4255 seconds
Time elapsed for test_boost: 14.2284 seconds

Time elapsed for test_std: 5.43268 seconds
Time elapsed for test_boost: 14.296 seconds
```

A bit surprised that boost performed so poorly. I am using the same pre-compiled binaries for Boost, which I understand is what the whole quasar ecosystem is using.

I also did a test using the Alma 9 (Boost 1.75) devel package, and the results were similar

```
Time elapsed for test_std: 5.35701 seconds
Time elapsed for test_boost: 13.9182 seconds

Time elapsed for test_std: 5.50923 seconds
Time elapsed for test_boost: 14.1848 seconds

Time elapsed for test_std: 5.35745 seconds
Time elapsed for test_boost: 14.0496 seconds
```

So, it is not a problem of our Boost version.

Reply:
> Actually, given that LogIt's main job is as part of Quasar servers, we should build a quick server on windows and linux and log from that. I can help to set this up with you

Yes, I will do

Reply:
Nice work - 3x faster is an unexpected benefit. Given the timing test you already carried out I'd say there is no need to go through the quasar server tests - I only thought this would be an easy way to validate and maybe a handy introduction to quasar. Seems to me the performance is validated, so we can do the quasar intro in another context outwith this MR.